spec/predicates/test-results.md

```diff
@@ -0,0 +1,148 @@
+# Predicate type: Test Results
+
+Type URI: https://in-toto.io/attestation/test-result/v0.1
+
+Version: 0.1.0
+
+Authors:
+
+## Purpose
+
+This predicate type defines a generic schema to express the result of running
+tests in software supply chains. The schema may be extended to support different
+types of testing or specific test harnesses.
+
+## Use Cases
+
+Software development processes include several types of tests. This attestation
+can be used to express the results of running those tests. It can be used to
+verify:
+
+1.  that all tests were in fact run, and
+2.  that all required tests passed
+
+Therefore, each attestation corresponds to one invocation of a test suite, and
+may include the results of several individual tests.
+
+## Prerequisites
+
+Understanding of the
+[in-toto attestation specification](https://github.com/in-toto/attestation).
+
+## Model
+
+This predicate type includes one compulsory field, `result`, that describes the
+result of the test run. The `testRun` object can be used to communicate a link
+to the test run and list tests that passed, failed, and passed with warnings. It
+also may optionally contain a reference to the configuration for the test run.
```

Comment:
What's the rationale for making the configuration reference optional?

It seems that having only the result with no indication of which tests were run may not be very helpful (e.g. it'd be good to know if the configured tests were from trusted config or untrusted config). I suppose this relates to my question above, about the potential use-cases for the predicate

Reply:
Continuing from comment about implementer, I made the configuration optional right now but I'm waiting to hear what parts they're willing to share with the results. I think there's value with configurations as well, so I'm okay to change this.

Also, do you have any thoughts on other things we'd include here?

Reply:
SGTM. Re: other things to include, I can't think of anything immediately -- IMO the reference to the test configuration and link to the test execution are the most important bits. It might be useful to consider the use-cases more explicitly, i.e. who is the consumer of test attestations, to see if any relevant info is missing (but feel free to ignore if there aren't concrete use-cases yet).

Reply:
Done!

Reply:
I just added some verification scenarios. Let me know if these make sense!

Reply:
Those all LGTM, thanks for the additional details!